How do I use query().map in 0.4?

Both in `0.4.0-beta.1` and on `master` I can't figure out how to use `query().map`, this works

```rust
use std::env;
use sqlx::postgres::{PgPool, PgRow};
use sqlx::Row;

struct Book {
    id: i32,
    title: String,
}

#[async_std::main]
async fn main() -> Result<(), sqlx::Error> {
    let pool = PgPool::connect(&env::var("DATABASE_URL").expect("env error!")).await?;

    sqlx::query("SELECT id, title FROM books")
        .try_map(|row: PgRow| {
            Ok(Book {
                id: row.get("id"),
                title: row.get("title"),
            })
        })
        .fetch_one(&pool)
        .await
        .expect("fetch error!");

    Ok(())
}
```

but this doesn't

```rust
use sqlx::postgres::{PgPool, PgRow};
use sqlx::Row;
use std::env;

struct Book {
    id: i32,
    title: String,
}

#[async_std::main]
async fn main() -> Result<(), sqlx::Error> {
    let pool = PgPool::connect(&env::var("DATABASE_URL").expect("env error!")).await?;

    sqlx::query("SELECT id, title FROM books")
        .map(|row: PgRow| Book {
            id: row.get("id"),
            title: row.get("title"),
        })
        .fetch_one(&pool)
        .await
        .expect("fetch error!");

    Ok(())
}
```

```rust
error[E0599]: no method named `fetch_one` found for struct `sqlx_core::query::Map<'_, sqlx_core::postgres::database::Postgres, impl sqlx_core::query::TryMapRow<sqlx_core::postgres::database::Postgres>, sqlx_core::postgres::arguments::PgArguments>` in the current scope
  --> src/main.rs:21:10
   |
21 |         .fetch_one(&pool)
   |          ^^^^^^^^^ method not found in `sqlx_core::query::Map<'_, sqlx_core::postgres::database::Postgres, impl sqlx_core::query::TryMapRow<sqlx_core::postgres::database::Postgres>, sqlx_core::postgres::arguments::PgArguments>`
   |
   = note: the method `fetch_one` exists but the following trait bounds were not satisfied:
           `<impl sqlx_core::query::TryMapRow<sqlx_core::postgres::database::Postgres> as std::ops::FnOnce<(sqlx_core::postgres::row::PgRow,)>>::Output = std::result::Result<_, sqlx_core::error::Error>`
           `impl sqlx_core::query::TryMapRow<sqlx_core::postgres::database::Postgres>: std::ops::Fn<(sqlx_core::postgres::row::PgRow,)>`
```


Both works fine in `0.3.5`

--- a/sqlx-core/src/query.rs
+++ b/sqlx-core/src/query.rs
@@ -356,13 +356,13 @@ where
 //
 // See https://github.com/rust-lang/rust/issues/62529
 
-pub trait TryMapRow<DB: Database> {
+pub trait TryMapRow<DB: Database>: Send {
     type Output: Unpin;
 
     fn try_map_row(&mut self, row: DB::Row) -> Result<Self::Output, Error>;
 }
 
-pub trait MapRow<DB: Database> {
+pub trait MapRow<DB: Database>: Send {
     type Output: Unpin;
 
     fn map_row(&mut self, row: DB::Row) -> Self::Output;
@@ -449,7 +449,7 @@ macro_rules! impl_map_row {
     ($DB:ident, $R:ident) => {
         impl<O: Unpin, F> crate::query::MapRow<$DB> for F
         where
-            F: FnMut($R) -> O,
+            F: Send + FnMut($R) -> O,
         {
             type Output = O;
 
@@ -460,7 +460,7 @@ macro_rules! impl_map_row {
 
         impl<O: Unpin, F> crate::query::TryMapRow<$DB> for F
         where
-            F: FnMut($R) -> Result<O, crate::error::Error>,
+            F: Send + FnMut($R) -> Result<O, crate::error::Error>,
         {
             type Output = O;
 

--- a/sqlx-core/src/query.rs
+++ b/sqlx-core/src/query.rs
@@ -251,7 +251,7 @@ where
 impl<'q, DB, F, O, A> Map<'q, DB, F, A>
 where
     DB: Database,
-    F: Send + Sync + Fn(DB::Row) -> Result<O, Error>,
+    F: TryMapRow<DB, Output = O>,
     O: Send + Unpin,
     A: 'q + Send + IntoArguments<'q, DB>,
 {
@@ -277,7 +277,7 @@ where
     /// Execute multiple queries and return the generated results as a stream
     /// from each query, in a stream.
     pub fn fetch_many<'e, 'c: 'e, E>(
-        self,
+        mut self,
         executor: E,
     ) -> BoxStream<'e, Result<Either<DB::Done, O>, Error>>
     where
@@ -294,7 +294,7 @@ where
                 r#yield!(match v {
                     Either::Left(v) => Either::Left(v),
                     Either::Right(row) => {
-                        Either::Right((self.mapper)(row)?)
+                        Either::Right(self.mapper.try_map_row(row)?)
                     }
                 });
             }
@@ -333,7 +333,7 @@ where
     }
 
     /// Execute the query and returns at most one row.
-    pub async fn fetch_optional<'e, 'c: 'e, E>(self, executor: E) -> Result<Option<O>, Error>
+    pub async fn fetch_optional<'e, 'c: 'e, E>(mut self, executor: E) -> Result<Option<O>, Error>
     where
         'q: 'e,
         E: 'e + Executor<'c, Database = DB>,
@@ -344,7 +344,7 @@ where
         let row = executor.fetch_optional(self.inner).await?;
 
         if let Some(row) = row {
-            (self.mapper)(row).map(Some)
+            self.mapper.try_map_row(row).map(Some)
         } else {
             Ok(None)
         }

